### PR TITLE
Added capacity

### PR DESCRIPTION
**Pull request by nomad-cc20**
Added course action capacity features:
- capacity attribute
- getter, setter
- bool hasCapacity() - "Is it possible to add another student?"
- corresponding unit tests
- capacity check in School::addStudentToCourseAction(...).

### CODE REVIEW

File: src/main/java/cz/upce/fei/inptp/zz/entity/School.java [hunk 1]
```diff
@@ -34,8 +32,10 @@ public boolean addStudentToCourseAction(Course course, Student student, TimeSlot
             return false;
         }
 
+        //check if specific timeSlot action is available in course
+        //check if course action has a free capacity for new student
         for (CourseAction action : course.getActions()) {
-            if (action.getTimeSlot().equals(timeSlot)) {
+            if (action.getTimeSlot().equals(timeSlot) && action.hasCapacity()) {
```
Comment (bLuma) [inline src/main/java/cz/upce/fei/inptp/zz/entity/School.java]: Maybe better check will be nested if statement. Now if capacity condition is not satisfied it results to looping through rest of actions in course list.

File: src/test/java/cz/upce/fei/inptp/zz/entity/SchoolTest.java [hunk 2]
```diff
@@ -206,4 +206,38 @@ public void testRemovedStudentWasRemovedFromHisCourseAction(){
        
         assertEquals(0, courseAction.getStudents().size());
     }
+    
+    @Test
+    public void testHasCapacityOnEmptyCourseAction() {
```
Comment (bLuma) [inline src/test/java/cz/upce/fei/inptp/zz/entity/SchoolTest.java]: All these tests are only CourseAction unit tests and shouldn't been placed in SchoolTest class

File: src/main/java/cz/upce/fei/inptp/zz/entity/CourseAction.java [hunk 3]
```diff
@@ -18,12 +18,25 @@
...
     private ActionType type;
+    private int capacity;
 
     public CourseAction() {
         students = new ArrayList<>();
-                
+        capacity = Integer.MAX_VALUE;
+    }
+    
+    public boolean hasCapacity() {
```
Comment (bLuma) [inline src/main/java/cz/upce/fei/inptp/zz/entity/CourseAction.java]: hasCapacity is something different from check that is inside of this method

**Review by bLuma — CHANGES_REQUESTED** [review]

File: src/main/java/cz/upce/fei/inptp/zz/entity/School.java (hunk 1 above)
Comment (nomad-cc20) [reply src/main/java/cz/upce/fei/inptp/zz/entity/School.java]: Solved.

File: src/main/java/cz/upce/fei/inptp/zz/entity/CourseAction.java (hunk 3 above)
Comment (nomad-cc20) [reply src/main/java/cz/upce/fei/inptp/zz/entity/CourseAction.java]: Solved.

File: src/test/java/cz/upce/fei/inptp/zz/entity/SchoolTest.java (hunk 2 above)
Comment (nomad-cc20) [reply src/test/java/cz/upce/fei/inptp/zz/entity/SchoolTest.java]: Solved.

**Comment by nomad-cc20** [comment]
Alright, the CourseActionTest class has been created and the corresponding tests were moved there. Also, the capacity check in AddStudentToCourseAction(...) is now nested. Plus, the method hasCapacity() got renamed to isNotFull().

**Review by bLuma — APPROVED** [review]

**Pull request merged**